app/controllers/mainCtrl.js

```diff
@@ -9,54 +9,74 @@
-                        ctrl.moneyInSell = result.rates[0].bid;
-
-                        ctrl.buy = parseFloat(ctrl.amountForBay * ctrl.money).toFixed(2);
-                        ctrl.sell = parseFloat(ctrl.amountForSell * ctrl.moneyInSell).toFixed(2);
-                    });
+            if (ctrl.amount === undefined) {
+                window.alert('Zbyt duza lub zbyt mala kwota startowa');
+            } else {
+                ctrl.wallet.pln = ctrl.amount;
+                ctrl.wallet.EUR = 0;
+                ctrl.wallet.USD = 0;
+                ctrl.wallet.CHF = 0;
+                ctrl.wallet.GBP = 0;
+                ctrl.wallet.AUD = 0;
+                ctrl.wallet.CAD = 0;
+                ctrl.wallet.HUF = 0;
+                ctrl.wallet.XDR = 0;
+            }
         };
 
         $localStorage.$default({
             wallet: {
-                pln: 0, eur: 0
+                pln: 0, USD: 0, EUR: 0, CHF: 0, GBP: 0, AUD: 0, CAD: 0, HUF: 0, XDR: 0
             }
         });
 
-        ctrl.confirmBuy = function ()
-        {
-
-            ctrl.buttonInBuy = ctrl.amountForBay - ctrl.wallet.pln;
-            console.log(ctrl.buttonInBuy);
-
-            ctrl.$storage = $localStorage.$default(ctrl.data.model = ctrl.buy);
-
-        };
-        ctrl.setWalletPln = function ()
-        {
-            ctrl.wallet.pln = ctrl.amount;
-        };
+        ctrl.wallet = Wallet.getWallet();
 
-        ctrl.buy = function (code, rate, value)
+        ctrl.getCurrency = function (selectedValue)
         {
-            ctrl.wallet[code] += value;
-            ctrl.wallet.pln -= rate * value;
-        };
+            CurrencyService.getCurrency(selectedValue)
+                    .then(function (result)
+                    {
+                        ctrl.money = result.rates[0].ask;
+                        ctrl.moneyInSell = result.rates[0].bid;
 
+                        ctrl.buyInWindow = parseFloat(ctrl.amountForBay * ctrl.money).toFixed(2);
+                        ctrl.sell = parseFloat(ctrl.amountForSell * ctrl.moneyInSell).toFixed(2);
 
-        ctrl.wallet = Wallet.getWallet();
+                        ctrl.buy = function ()
+                        {
+                            if (ctrl.amountForBay === undefined) {
+                                window.alert('Zbyt duza lub zbyt mala kwota startowa');
+                            } else if (Math.round(ctrl.wallet.pln * 100) < Math.round(ctrl.buyInWindow * 100)) {
+                                window.alert('Za malo pieniedzy :(');
+                            } else if (Math.round(ctrl.wallet.pln * 100) >= Math.round(ctrl.buyInWindow * 100)) {
+                                ctrl.wallet[ctrl.data.model] += ctrl.amountForBay;
+                                ctrl.wallet.pln -= ctrl.money * ctrl.amountForBay;
+                            }
+                        };
+
+                        ctrl.seller = function ()
```

Comment:
You have to be consistent in naming. If you create `buy`function you should create `sell` function